File: src/mantle/world/DimensionCoordPair.scala
package mantle.world

import java.lang.String

/**
 * DimensionCoordPair
 *
 * @param dim The dimension ID
 * @param x The x coordinate
 * @param z The z coordinate
 *
 * @author mDiyo, Sunstrike
 */
class DimensionCoordPair(val dim: Int, val x:Int, val z:Int) {

  def this(tuple: DimensionCoordPair) = this(tuple.dim, tuple.x, tuple.z)

  def equalCoords(dim: Int, posX: Int, posZ: Int): Boolean = this.dim == dim && this.x == posX && this.z == posZ

  override def hashCode: Int = {
    val prime: Int = 31
    var result: Int = 1
    result = prime * result + dim
    result = prime * result + x
    result = prime * result + z

    result
  }

  override def equals(obj: scala.Any): Boolean = obj match {
    case a:DimensionCoordPair => equalCoords(a.dim, a.x, a.z)
    case _                    => false
  }

  override def toString: String = s"Dim: $dim, X: $x, Z: $z"

}

File: src/mantle/world/DimensionCoordPair.java
package mantle.world;

/**
 * DimensionCoordPair
 *
 * @author mDiyo
 */
public class DimensionCoordPair
{
    public final int dim;

    public final int x;

    public final int z;

    public DimensionCoordPair(int worldID, int posX, int posZ)
    {
        this.dim = worldID;
        this.x = posX;
        this.z = posZ;
    }

    public boolean equalCoords(int worldID, int posX, int posZ)
    {
        return this.dim == posX && this.x == posX && this.z == posZ;
    }

    @Override
    public boolean equals(Object obj)
    {
        if (obj == null)
        {
            return false;
        }

        if (this.getClass() == obj.getClass())
        {
            DimensionCoordPair coord = (DimensionCoordPair) obj;
            if (this.dim == coord.dim && this.x == coord.x && this.z == coord.z)
            {
                return true;
            }
        }
        return false;
    }

    @Override
    public int hashCode()
    {
        final int prime = 37;
        int result = 1;
        result = prime * result + this.dim;
        result = prime * result + this.x;
        result = prime * result + this.z;
        return result;
    }

    @Override
    public String toString()
    {
        return "Dim: " + this.dim + ", X: " + this.x + ", Z: " + this.z;
    }
}
